src/client/components/pages/Courses/SemesterTable.tsx

```diff
@@ -0,0 +1,105 @@
+import React, {
+  FunctionComponent,
+  ReactElement,
+} from 'react';
+import {
+  Table,
+  TableHead,
+  TableRow,
+  TableHeadingCell,
+  TableCell,
+  TableHeadingSpacer,
+  Checkbox,
+  TableBody,
+  Form,
+} from 'mark-one';
+import { COURSE_TABLE_COLUMN, COURSE_TABLE_COLUMN_GROUP, MANDATORY_COLUMNS } from 'common/constants';
+import { ModalFieldsColumn } from './modalFields';
+
+interface SemesterTableProps {
+  /**
+   * List of columns that should be displayed for show/hide
+   */
+  columns: ModalFieldsColumn[],
+
+  /**
+   * Of the columns above, the unique column reference to columns that should
+   * be considered visible (or the checkbox checked). Columns present in
+   * `columns`, but not here will be displayed unchecked.
+   */
+  checked: COURSE_TABLE_COLUMN[],
+
+  /**
+   * Handler to be invoked when a checkbox is checked or unchecked
+  */
+  onChange: (cols: COURSE_TABLE_COLUMN) => void;
+}
+
+/**
+ * Component to show/hide columns in the CourseInstance cable
```

Comment:
I think that you meant to say "table" instead of "cable" here.

Reply:
Yes....yes I did. :laughing: 

Fixed in 99b016345a77702190f894b7c3e768254bba98ce